src/OrchardCore.Modules/OrchardCore.Tenants/Workflows/Activities/CreateTenantTask.cs

```diff
@@ -100,10 +96,10 @@ public async override Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecuti
                 shellSettings["DatabaseProvider"] = databaseProviderTask.Result?.Trim();
                 shellSettings["Secret"] = Guid.NewGuid().ToString();
                 shellSettings["RecipeName"] = recipeNameTask.Result.Trim();
+
+                ShellSettingsManager.SaveSettings(shellSettings);
```

Comment:
You can remove this line, this is done by `UpdateShellSettingsAsync()` internally.

Note: Just a 1st review before merging #4164 on dev and rebase your PR